tools/interactive/interactivetool_climate_notebook.xml

```diff
@@ -11,7 +11,7 @@
     <environment_variables>
         <environment_variable name="HISTORY_ID">$__history_id__</environment_variable>
         <environment_variable name="REMOTE_HOST">$__galaxy_url__</environment_variable>
-        <environment_variable name="GALAXY_WEB_PORT">8080</environment_variable>
+        <environment_variable name="GALAXY_WEB_PORT">${__app__.config.galaxy_infrastructure_web_port}</environment_variable> <!-- FIXME: Warning: The use of __app__ is deprecated and will break backward compatibility in the near future -->
```

Comment:
That's a weird statement, why ?

Reply:
You mean the comment, right?

It was copied over from the RStudio tool xml, but the comment might be out of date and no plans to remove app support anymore. Just did not want to remove a comment like that.

Reply:
Yeah, can you remove the comment in the tools you've modified ?

I don't like particularly like GALAXY_WEB_PORT, we do already have GALAXY_URL that .... should of course include any non-default port. Do you know if this is being used by any IT ?

Reply:
I can remove the comments yes. I don't know if the web port is used anywhere. Unfortunately one would need to look at all the Dockerfiles to be sure. But probably not. It is not used in RStudio so I removed it from there.

I just though it at least did not make any sense to hardcode this to 8080 everywhere. This is in any case not an important point for the rest of the PR, just an attempt to clean things up at the same time.

I don't know why there is a `galaxy_infrastructure_web_port`either...

Reply:
(like on our repo, not in the tools)

Reply:
Done. Commit is gone now

Reply:
Actually, I misremembered. It is used in RStudio due to this: https://github.com/bgruening/galaxy_ie_helpers/blob/378f0216882a279e35ceda880c2038147fdd6db9/galaxy_ie_helpers/__init__.py#L82

But that logic does not match what you are saying, since the `GALAXY_URL` should contain the port. However the code fails without the `GALAXY_WEB_PORT` and 8080 is incorrect in many cases. So there was a real reason why I did this for RStudio. It would fail otherwise in my setup without changing the tool XML which is not very nice.

Any thoughts?

Reply:
That code is very old, it needs an update ... can you use cheetah to grab the port out of `$__galaxy_url__` ?

Reply:
(just for rstudio)

Reply:
Ok, I needed to dig a bit into my memory and test a bit. Here is the full story:

When trying to run galaxy using `galaxy_infrastructure_url: http://localhost:3456` i get this when trying to connect from RStudio to Galaxy:

```
> gx_save()
DEBUG:root:Host IP determined to be b'10.0.2.2\n'
DEBUG:root:TestURL url=http://localhost:3456 obj=True
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): localhost:3456
DEBUG:root:TestURL url=http://localhost:3456 state=failure
DEBUG:root:TestURL url=http://b'10.0.2.2':8080 obj=True
DEBUG:urllib3.connectionpool:Starting new HTTP connection (1): b'10.0.2.2':8080
DEBUG:root:TestURL url=http://b'10.0.2.2':8080 state=failure
Traceback (most recent call last):
  File "/usr/local/bin/put", line 17, in <module>
    put( args.filepath, file_type=args.filetype, history_id=args.history_id )
  File "/opt/miniconda/lib/python3.9/site-packages/galaxy_ie_helpers/__init__.py", line 113, in put
    gi = get_galaxy_connection(history_id=history_id)
  File "/opt/miniconda/lib/python3.9/site-packages/galaxy_ie_helpers/__init__.py", line 100, in get_galaxy_connection
    raise Exception(msg)
Exception: Could not connect to a galaxy instance. Please contact your SysAdmin for help with this error
```

So the first try (using `GALAXY_URL`) directly fails due to the fact that `localhost` inside the docker container does not point to the host. So the backup code is there to pick up the ip to the host (which I believe is configured as the router inside the container, or something like that), however then the `GALAXY_WEB_PORT` is used to get the port number instead of the `GALAXY_URL`. And this is hardcoded to 8080 everywhere.

Since this probably only happens in developer setups, where 99% of users are using "http://localhost:8080" anyway, hardcoding "8080" is not that bad. And granted, using another port for a developer setup is probably asking for trouble (however, if you are asking for trouble, you have come to the right place! 😁). Still, to me it seemed more correct to get this port from the config. Also, I don't think there is a reason to fix this only for RStudio. For instance EtherCalc is also making use of the same old `galaxy_ie_helpers`: https://hub.docker.com/r/shiltemann/ethercalc-galaxy-ie/dockerfile

So using cheetah as you suggest is probably the best fix, if this is to be fixed. But then again, this is a very slight problem that will only appears for developers that are using an unstandard port, and in those few cases, they can just temporarily change the tool xml. In all other cases, the current solution works. Someone should rather update the `galaxy_ie_helpers` and then the tools that use that library, but that is definitely another PR.

I believe the reason I had trouble making this work in my case was due to Podman and not the `GALAXY_WEB_PORT`. That the wrong port was used in the backup solution was just something I noticed during debugging. And in any case, if you are using Podman for development, you are also asking for trouble, so I gave up fixing that bug. Callback from a container to Galaxy on a standard Docker setup on your developer laptop should work, as I haven't changed anything related to this.

So a very long story short: I suggest we just leave this out. The above description might be useful for someone sometime.